<a href="https://colab.research.google.com/github/pathilink/sp_global_bootcamp/blob/main/5_exercicio_teste_hipotese/exercicio_teste_hipotese.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Exercício Teste de Hipóteses

## Bibliotecas

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

## Dados

### Experimento

In [2]:
df_experimento = pd.read_csv('/content/drive/MyDrive/WoMakersCode/Bootcamp S&P Global/Teste de Hipótese/experimento_teste_ab.csv')
df_experimento.head()

Visitante_ID Versão_Página  Visualizações  Conversões
0             1             A            167           1
1             2             A            160           0
2             3             A            170           0
3             4             A            161           1
4             5             A            104           0

In [3]:
df_experimento.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Visitante_ID   2000 non-null   int64 
 1   Versão_Página  2000 non-null   object
 2   Visualizações  2000 non-null   int64 
 3   Conversões     2000 non-null   int64 
dtypes: int64(3), object(1)
memory usage: 62.6+ KB


In [4]:
# converter colunas numéricas para categóricas
df_experimento['Visitante_ID'] = df_experimento['Visitante_ID'].astype('object')
# df_experimento['Conversões'] = df_experimento['Conversões'].astype('object')

In [5]:
df_experimento[['Visitante_ID', 'Conversões']].dtypes


Visitante_ID    object
Conversões       int64
dtype: object

In [6]:
df_experimento['Visualizações'].describe()

count    2000.000000
mean      150.755500
std        28.809421
min       100.000000
25%       126.000000
50%       151.000000
75%       175.000000
max       199.000000
Name: Visualizações, dtype: float64

In [7]:
df_experimento['Versão_Página'].unique()

array(['A', 'B'], dtype=object)

In [8]:
df_experimento['Conversões'].unique()

array([1, 0])

### Pacientes

In [9]:
df_pacientes = pd.read_csv('/content/drive/MyDrive/WoMakersCode/Bootcamp S&P Global/Teste de Hipótese/pacientes.csv', index_col=0)
df_pacientes.head()

Pressao_Arterial  Idade  Genero  Etnia  Estado_Saude Nome_Genero  \
0        142.755809     31       1      1             1    Feminino   
1        161.878468     35       1      1             0    Feminino   
2        161.508259     39       0      3             1   Masculino   
3        144.979408     43       1      3             1    Feminino   
4        162.502954     55       0      2             1   Masculino   

       Nome_Etnia                  Nome_Estado_Saude  
0  Afro-americano  Com condições de saúde adicionais  
1  Afro-americano                           Saudável  
2        Asiático  Com condições de saúde adicionais  
3        Asiático  Com condições de saúde adicionais  
4       Hispânico  Com condições de saúde adicionais

## Questões - Teste AB

### 1.
Qual dos cenários tem a maior taxa de conversão?

In [10]:
df_experimento.groupby('Versão_Página')['Conversões'].count()

Versão_Página
A    1000
B    1000
Name: Conversões, dtype: int64

In [11]:
df_experimento.groupby('Versão_Página')['Conversões'].sum()

Versão_Página
A    105
B    125
Name: Conversões, dtype: int64

<font color=blue>
A página com a versão B apresentou a maior taxa de conversão: 12,5%.
</font>

### 2.
Calcule qual o tamanho da amostra necessária para o desenvolvimento
de um teste A/B, seguindo os seguintes critérios:

a. O cenário A, da base, como o inicial, que funciona hoje.

b. Considere 95% de confiança de que o efeito na conversão não foi aleatório com um nível de significância de 5% (alpha = 0,05).

c. Também considere 80% de certeza conseguir capturar o efeito da nova abordagem.

d. O aumento para 10% de conversão.

## Questões - Pacientes

Considere a base como a população, portanto as estatísticas da população são conhecidas.

Obs.: Como o dado Pressao_Arterial está sendo utilizado como um valor único, podemos supor que ou é a PAM ou um valor fixo (sistólica ou diastólica).

### 3.

Considerando uma amostra de 45 números que representam o index do dataframe, índices= ([909, 751, 402, 400, 726, 39, 184, 269, 255, 769, 209, 715, 677, 381, 793, 697, 89, 280, 232, 756, 358, 36, 439, 768, 967, 699, 473, 222, 89, 639, 883, 558, 757, 84, 907, 895, 217, 224, 311, 348, 146, 505, 273, 957, 362]). Considerando essa amostra é possível dizer que a idade média das pessoas com problemas cardíacos é maior que 50 anos? Nível de significância igual a 5%.

### 4.
Queremos entender que tipo de amostra estamos lidando se dividirmos os conjuntos em 2, sendo um com pessoas que têm condições de saúde adicionais e o outro com pessoas saudáveis. Seria dependente ou independente?

### 5.
Agora considere o um conjunto de pessoas aleatória que representam o index do dataframe, índices = ([690, 894, 67, 201, 364, 19, 60, 319, 588, 643, 855, 623, 530, 174, 105, 693, 6, 462, 973, 607, 811, 346, 354, 966, 943, 372]), podemos dizer que a pressão arterial média para
pacientes com condições de saúde adicionais é igual à pressão arterial média para pacientes sem condições adicionais de saúde? Considere o nível de significância a 6%.

### 6.
Existe uma diferença significativa na pressão arterial média entre diferentes grupos étnicos nesta população? (Teste ANOVA, alpha é 5%)

a. Hipótese Nula (H0): A pressão arterial média é a mesma em todos os grupos étnicos.

b. Hipótese Alternativa (H1): Há uma diferença significativa na pressão arterial média entre pelo menos dois grupos étnicos nesta população.

### 7.
Lúcia é uma pesquisadora e tem o objetivo de entender a relação de gênero neste grupo de pacientes. Acredita-se que há uma relação entre o sexo e condições de saúde adicionais. (Teste qui-quadrado)

### 8.
Existe uma associação entre a idade dos pacientes e sua pressão arterial?

a. Hipótese Nula (H0): se a pressão arterial é independente da idade

b. Hipótese Alternativa (H1): a pressão arterial esta associada a idade

### 9.
Qual é o intervalo de confiança para a média da pressão arterial entre os pacientes com condições de saúde adicionais? (nível de confiança 95%)

### 10.
A distribuição da pressão arterial na população segue uma distribuição normal?

a. Hipótese Nula (H0): A distribuição da pressão arterial na população segue uma distribuição normal.

b. Hipótese Alternativa (H1): A distribuição da pressão arterial na população não segue uma distribuição normal.